src/providers/crypto/cryptors/aes_cbc.rs

```diff
@@ -0,0 +1,205 @@
+//! # AES-CBC cryptor module.
+//!
+//! Module contains [`AesCbcCryptor`] type which can be used for data encryption
+//! and decryption.
+
+use aes::cipher::{block_padding::Pkcs7, BlockDecryptMut, BlockEncryptMut, KeyIvInit};
+use sha2::{Digest, Sha256};
+
+use crate::{
+    core::{Cryptor, EncryptedData, PubNubError},
+    lib::alloc::{format, string::ToString, vec, vec::Vec},
+};
+
+type Encryptor = cbc::Encryptor<aes::Aes256>;
+/// AES-SHA256 encryptor type.
+type Decryptor = cbc::Decryptor<aes::Aes256>;
+
+/// Unique cryptor identifier
+const IDENTIFIER: [u8; 4] = *b"ACRH";
+
+/// AES cipher block size.
+const AES_BLOCK_SIZE: usize = 16;
+
+/// AES-CBC cryptor.
+#[derive(Debug)]
+pub struct AesCbcCryptor {
+    /// Key for data _encryption_ / _decryption_.
+    cipher_key: Vec<u8>,
+}
+
+impl AesCbcCryptor {
+    /// Create AES-CBC cryptor.
+    pub fn new<K>(cipher_key: K) -> Result<Self, PubNubError>
+    where
+        K: Into<Vec<u8>>,
+    {
+        let cipher_key = cipher_key.into();
+
+        if cipher_key.is_empty() {
+            return Err(PubNubError::CryptoInitialization {
+                details: "Cipher key is empty".into(),
+            });
+        }
+
+        Ok(Self {
+            cipher_key: Self::sha256(cipher_key),
+        })
+    }
+
+    fn initialization_vector(&self) -> [u8; 16] {
```

Comment:
```suggestion
    fn initialization_vector(&self) -> [u8; AES_BLOCK_SIZE] {
```
just for consistency